src/core/server/saved_objects/service/lib/search_dsl/query_params.ts

```diff
@@ -39,17 +39,27 @@ function getTypes(mappings: IndexMapping, type?: string | string[]) {
 }
 
 /**
- *  Get the field params based on the types and searchFields
+ *  Get the field params based on the types, searchFields, and rootSearchFields
  */
-function getFieldsForTypes(types: string[], searchFields?: string[]) {
-  if (!searchFields || !searchFields.length) {
+function getFieldsForTypes(
+  types: string[],
+  searchFields: string[] = [],
+  rootSearchFields: string[] = []
```

Comment:
In other places the repository uses a pattern of `type.attributes.field` vs `type.field` to distinguish between root fields/type fields. It would be more consistent if we could do it here too (even though it's a breaking change).

This inconsistency wasn't introduced by this PR, but it highlights the current shortcomings. So just leaving this as a note to the Platform team that this is something we should take a look at at some point.